### ISSUE TITLE
General improvements to woocommerce/components

### ISSUE THREAD

**Issue by samueljseay** [woocommerce/woocommerce-admin#7297]
This epic is to track work needed to improve `@woocommerce/components`.

Discussion here: p90Yrv-2k8-p2

- [ ] Documentation - introduce longform MDX docs alongside Storybook stories. Finish adding args support to the existing component stories

- [ ] Explore css-in-js solution (start with Linaria). First step is spike/experimentation. Deliverable would be a doc with some findings and some demo code.

- [ ] TypeScript conversion, start by migrating low-effort components to TS and also publish the type defs with the package.

**Comment by mattsherman** [comment]
@joelclimbsthings What is the status of this? You were exploring the css-in-js solution, right? A P2 post with your findings would be excellent. And we should put the rest of his on the backburner until the next cool down (we can also consider some of this work for a future cycle).

**Comment by joelclimbsthings** [comment]
@mattsherman Yup! I spent some time looking into alternatives to Linaria as well. Planning to put up a P2, although I didn't get as far with this as I'd like due to portering and other tasks, so I plan to finish it off during the next cool down.
